DOCS-4640 APM troubleshooting doc update (#16547)
DataDog/documentation: content/en/tracing/troubleshooting/_index.md

    Knowing what versions are being used allows us to ensure integrations are supported in our [Compatiblity Requirements][15] section, check for known issues, or to recommend a tracer or language version upgrade if it will address the problem.

> ## Service naming convention issues

Reviewer: I would move this section above Troubleshooting data requested by Datadog Support

> If you run into quota-related issues, try following the steps below to ensure you're following best practices for service naming conventions.

Reviewer: ```suggestion
If you run into quota-related issues, try following these best practices for service naming conventions.
```

### Exclude environment tag values from service names

> By default, the environment (`env`) is the primary tag for Datadog APM.

Reviewer: May be worth linking to this APM guide here: https://docs.datadoghq.com/tracing/guide/setting_primary_tags_to_scope/.

{{< img src="/tracing/troubleshooting/troubleshooting-service-naming-convention-issues-3.png" alt="Environment is the default primary tag" style="width:100%;" >}}

A service is typically deployed in multiple environments, such as `prod`, `staging`, and `dev`. Performance metrics like request counts, latency, and error rate differ across various environments. The environment dropdown in the Service Catalog allows you to scope the data in the **Performance** tab to a specific environment.

{{< img src="/tracing/troubleshooting/troubleshooting-service-naming-convention-issues-2.png" alt="Choose a specific environment using the `env` dropdown in the Service Catalog" style="width:100%;" >}}

> One pattern that often leads to issues with an overwhelming number of services is to include the environment value in service names. For example, you might have two unique services instead of one since they are operating in two separate environments: `prod-web-store` and `dev-web-store`.

Reviewer: ```suggestion
One pattern that often leads to issues with an overwhelming number of services is including the environment value in service names. For example, you might have two unique services instead of one since they are operating in two separate environments: `prod-web-store` and `dev-web-store`.
```

> You should avoid this pattern and rename your services by tuning your instrumentation.

Reviewer: ```suggestion
Datadog recommends tuning your instrumentation by renaming your services.
```

> **Note**: Datadog's trace metrics are unsampled, which means the app shows all data instead of subsections of them. The [volume guidelines][17] are applied to ensure that we consistently provide users with an optimal experience across the app.

Reviewer: The second sentence in this note sounded internal facing to me, feel free to reword.

```suggestion
Trace metrics are unsampled, which means your instrumented application shows all data instead of subsections of them. The [volume guidelines](#data-volume-guidelines) are also applied. 
```

### Use the second primary tag instead of putting metric partitions or grouping variables into service names

Second primary tags are additional tags that you can use to group and aggregate your trace metrics. You can use the dropdown to scope the performance data to a given cluster name or data center value.

{{< img src="/tracing/troubleshooting/troubleshooting-service-naming-convention-issues-1.png" alt="Use the dropdown menu to select a specific cluster or data center value" style="width:100%;" >}}

> Including metric partition or grouping variables in service names instead of applying the second primary tag will unnecessarily inflate the number of unique services in an account and result in potential delay or data loss. For example, instead of the service `web-store`, you might decide to name different instances of a service `web-store-us-1`, `web-store-eu-1`, and `web-store-eu2` to see performance metrics for these partitions side by side. Instead, we recommend implementing the **region value** as a second primary tag.

Reviewer: ```suggestion
Including metric partitions or grouping variables in service names instead of applying the second primary tag unnecessarily inflates the number of unique services in an account and results in potential delay or data loss.

 For example, instead of the service `web-store`, you might decide to name different instances of a service `web-store-us-1`, `web-store-eu-1`, and `web-store-eu2` to see performance metrics for these partitions side-by-side. Datadog recommends implementing the **region value** as a second primary tag.
```

[14]: /tracing/custom_instrumentation/
[15]: /tracing/compatibility_requirements/
[16]: /tracing/guide/setting_primary_tags_to_scope/?tab=helm#add-a-second-primary-tag-in-datadog

> [17]: /tracing/troubleshooting/#data-volume-guidelines

Reviewer: ```suggestion
```
